lib/unison-sqlite/src/Unison/Sqlite/DB.hs

```diff
@@ -0,0 +1,272 @@
+  (Maybe b -> Either e r) ->
+  m r
+queryMaybeCheck s params check = do
+  conn <- ask
+  liftIO (Connection.queryMaybeCheck conn s params check)
+
+queryMaybeOneCheck ::
+  (DB m, Sqlite.FromField b, Sqlite.ToRow a, Show e, Typeable e) =>
+  Sql ->
+  a ->
+  (Maybe b -> Either e r) ->
+  m r
+queryMaybeOneCheck s params check = do
+  conn <- ask
+  liftIO (Connection.queryMaybeOneCheck conn s params check)
+
+queryOneCheck ::
+  (DB m, Sqlite.FromRow b, Sqlite.ToRow a, Show e, Typeable e) =>
+  Sql ->
+  a ->
+  (b -> Either e r) ->
+  m r
+queryOneCheck s params check = do
+  conn <- ask
+  liftIO (Connection.queryOneCheck conn s params check)
+
+queryOneOneCheck ::
+  (DB m, Sqlite.FromField b, Sqlite.ToRow a, Show e, Typeable e) =>
+  Sql ->
+  a ->
+  (b -> Either e r) ->
+  m r
+queryOneOneCheck s params check = do
+  conn <- ask
+  liftIO (Connection.queryOneOneCheck conn s params check)
+
+-- With results, without parameters, without checks
+
+queryList_ :: (DB m, Sqlite.FromRow a) => Sql -> m [a]
+queryList_ s = do
+  conn <- ask
+  liftIO (Connection.queryList_ conn s)
+
+queryListOne_ :: (DB m, Sqlite.FromField a) => Sql -> m [a]
+queryListOne_ s = do
+  conn <- ask
+  liftIO (Connection.queryListOne_ conn s)
+
+queryMaybe_ :: (DB m, Sqlite.FromRow a) => Sql -> m (Maybe a)
+queryMaybe_ s = do
+  conn <- ask
+  liftIO (Connection.queryMaybe_ conn s)
+
+queryMaybeOne_ :: (DB m, Sqlite.FromField a) => Sql -> m (Maybe a)
+queryMaybeOne_ s = do
+  conn <- ask
+  liftIO (Connection.queryMaybeOne_ conn s)
+
+queryOne_ :: (DB m, Sqlite.FromRow a) => Sql -> m a
+queryOne_ s = do
+  conn <- ask
+  liftIO (Connection.queryOne_ conn s)
+
+queryOneOne_ :: (DB m, Sqlite.FromField a) => Sql -> m a
+queryOneOne_ s = do
+  conn <- ask
+  liftIO (Connection.queryOneOne_ conn s)
+
+-- With results, without parameters, with checks
+
+queryListCheck_ :: (DB m, Sqlite.FromRow a, Show e, Typeable e) => Sql -> ([a] -> Either e r) -> m r
+queryListCheck_ s check = do
+  conn <- ask
+  liftIO (Connection.queryListCheck_ conn s check)
+
+queryListOneCheck_ :: (DB m, Sqlite.FromField a, Show e, Typeable e) => Sql -> ([a] -> Either e r) -> m r
+queryListOneCheck_ s check = do
+  conn <- ask
+  liftIO (Connection.queryListOneCheck_ conn s check)
+
+queryMaybeCheck_ :: (DB m, Sqlite.FromRow a, Show e, Typeable e) => Sql -> (Maybe a -> Either e r) -> m r
+queryMaybeCheck_ s check = do
+  conn <- ask
+  liftIO (Connection.queryMaybeCheck_ conn s check)
+
+queryMaybeOneCheck_ :: (DB m, Sqlite.FromField a, Show e, Typeable e) => Sql -> (Maybe a -> Either e r) -> m r
+queryMaybeOneCheck_ s check = do
+  conn <- ask
+  liftIO (Connection.queryMaybeOneCheck_ conn s check)
+
+queryOneCheck_ :: (DB m, Sqlite.FromRow a, Show e, Typeable e) => Sql -> (a -> Either e r) -> m r
+queryOneCheck_ s check = do
+  conn <- ask
+  liftIO (Connection.queryOneCheck_ conn s check)
+
+queryOneOneCheck_ :: (DB m, Sqlite.FromField a, Show e, Typeable e) => Sql -> (a -> Either e r) -> m r
```

Comment:
Rather than using `One` for both "row vs field" and "single row vs multiple rows" we could use separate names for these separate concepts.

I'd suggest: `Single` vs `Multi` for whether it returns a single row vs many rows
and `Row vs Field` for the other dimension.

So then we would have:

* `querySingleField`
* `querySingleRow`
* `queryMultiField`
* `queryMultiRow`

Currently, If I just see `queryOneCheck` I'm not sure whether it's `one` field or `one` row.

Reply:
Did you see this comment? It answers the question of "what does `queryOneCheck` mean.

https://github.com/unisonweb/unison/pull/2668/files#diff-780c176d18fc11cc4f5fef342177cb250f84e6ab54c22569eaf94fd10660f860R106-R128

And is this the alternative naming scheme you are proposing?

```
query(Multi|Maybe|Single)(Row|Field)[Check][_]
```

Reply:
Yes I saw that, thank you for writing it, my concern is that the comment won't be clearly visible at usage sites so I'd like the names to be as clear as possible at a glance, and I think having `One` mean something in both dimensions contributes just a smidge of unnecessary confusion 😄 

Yes, that'd be my proposed alternative naming scheme 👍🏼

Reply:
I hear you about the overloaded `One`/`One`. I'd also like to resolve it.

I'm not sure `Multi` is a good name change from `List`; it kind of closes the door to a variant that returns a `Vector`, for example. I think `List`/`Maybe`/`One` was okay for that.

Perhaps just changing this part would get us there?

Current:
- No suffix means 2+ column row
- `One` suffix means 1 column row
  
Proposed:
- `Row` suffix means 2+ column row
- `Col` suffix means 1 column row

I just changed `Field` to `Col`; I think "column" is a more established/standard name than "field" for these things.

So:

```
query(List|Maybe|One)(Row|Col)[Check][_]
```

Reply:
Looks good to me 👍🏼  Thanks!